### PR TITLE
Added 'ranking' meta-analysis (bucket-level multi-model ranking)

### PR DESCRIPTION
**Pull request by hwidjaja**

### CODE REVIEW

File: explainaboard/meta_analyses/ranking.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+import numpy as np
+import pandas as pd
+
+from explainaboard.info import SysOutputInfo
+from explainaboard.meta_analyses.utils import report_to_sysout
+from explainaboard.utils.typing_utils import unwrap
+
+
+class RankingMetaAnalysis:  # (can inherit from an abstract MetaAnalysis class)
```
Comment (pfliu-nlp) [inline explainaboard/meta_analyses/ranking.py:13]: It seems that there is some code redundancy between classes `RankingMetaAnalysis` and `RankFlippingMetaAnalysis`. We can try to abstract these two classes later. Two hints:
* (1) define a base class from which RankingMetaAnalysis and RankFlippingMetaAnalysis can inherit
* (2) define a general class that can cover three situations: 1 system, 2 systems, and multiple systems.

**Review by pfliu-nlp — APPROVED** [review]
it looks good overall

**Comment by pfliu-nlp** [comment]
Hi, @neubig  would like to have a quick review of this PR?

**Comment by neubig** [comment]
Please go ahead, I don't want to hold this up while I'm on vacation.

**Pull request merged**